### DISCUSSION TITLE
Welcome to geomodelgrids Discussions!

### DISCUSSION THREAD

**Discussion by baagaard-usgs** [baagaard-usgs/geomodelgrids#130] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by HeasmTehrani** [comment]
Hi,

I am working on a USGS project in developing shear wave velocity profile model.
I would like to use the model, geomodelgrids_borehole, to derive the information related to specific points to have a comparison.
The problem is that when I want to use the model in Spyder to automatically download the data for all points, I face an error "sh: 1: /geomodelgrids_borehole: not found", although I am in the correct directory and it is working in the terminal.
Can you help me with solving this issue?

Thanks,
Hesam

**Comment by baagaard-usgs** [comment]
I am not sure how you are trying to use`geomodelgrids_borehole` with Spyder. `geomodelgrids_borehole` is a command line program that is normally run in a terminal or shell. If you downloaded the binary and unpacked the binary distribution, then you should be able to run the program in a terminal window. Make sure you run `source setup.sh` in the top-level directory as indicated in the instructions https://geomodelgrids.readthedocs.io/en/latest/getting-started/install.html#installing-the-binary-package _before_ attempting to run the program.

**Comment by HeasmTehrani** [comment]
Hi Brad,

Hope you are perfect.
I did all steps completely and the model works perfectly fine when I am using the terminal.
But my problem is that I need to run it for thousands of locations; thus, I tried to write a simple code in pyhon to automatically derive the information for the desired locations. This is my code attached:
import os
import pandas as pd

data=pd.read_excel("/home/hsalmanitehrani/Downloads/LatLon.xlsx", dtype='str')
data["Lat"] = data["Lat"].astype('float')
data["Lon"] = data["Lon"].astype('float')

fpath = "/home/hsalmanitehrani/geomodelgrids/geomodelgrids-1.0.0rc2-Linux_x86_64"
os.chdir(fpath)
os.system(". ./setup.sh")

fpath = "./src/tests/data"
os.chdir(fpath)

for i in range(len(data)):
    a = os.system("geomodelgrids_borehole --models=USGS.h5  --location=%f,%f --max-depth=2000 --dz=1 --output=USGS/%s.out --values=Vs" %(data.Lat[i],data.Lon[i],str(data.Linux_name[i])))
    if a!=0:
        print(i,str(data.Linux_name[i]))

The problem is that when I run it, I get an error: "sh: 1: /geomodelgrids_borehole: not found", although in terminal, this line works well and I get response for a specific location.
Do you have any suggestions that help me solve this issue?
Thanks for your time and consideration.

Best,
Hesam
________________________________

**Comment by baagaard-usgs** [comment]
You need to setup the shell environment spawned by Python the same way that you do when you run `geomodelgrids_borehole` in a terminal.

I suggest using [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) instead of `os.system()`. You can pass a dictionary with environment variables to the `env` argument of `subprocess.run()`. This allows you to set the PATH environment variable so that the `geomodelgrids_borehole` executable will be found. You may also need to set the `LD_LIBRARY_PATH` environment variable if you are using Linux. The `setup.sh` script will show what variables need to be set.
